tox.ini

```diff
@@ -3,50 +3,59 @@ envlist = py, lint, fmt
 
 [testenv]
 description = run unit tests
-deps =
-    pytest>=7
+allowlist_externals=
+    pytest
+    poetry
+commands_pre =
+    poetry install --with dev --no-root
 commands =
+    pip freeze
     pytest {posargs:tests}
 
 [testenv:fmt]
 description = format with pre-commit
-deps =
-    pre-commit
-commands = ./scripts/fmt.sh
-allowlist_externals = ./scripts/fmt.sh
+commands_pre =
+    poetry install --with dev --no-root
+commands =
+    ./scripts/fmt.sh
+allowlist_externals =
+    poetry
+    ./scripts/fmt.sh
 
 [testenv:lint]
 description = lint with pylint
-deps =
-    pylint>=2.16.2,<=3.1.0
-    pytest
-    .[dev]
+commands_pre =
+    poetry install --with dev --no-root
 commands = pylint tuning scripts/*.py build/*.py tests
-allowlist_externals = pylint
+allowlist_externals =
+    poetry
+    pylint
 
 [testenv:build]
 description = build wheel
-deps =
-    build
-commands = python -m build -w
+allowlist_externals=
+    poetry
+commands = poetry build --format wheel
 skip_install = True
 
 [testenv:twinecheck]
 description = check wheel
+commands_pre = []
 deps =
     twine
 commands = twine check dist/*
 skip_install = True
 
 [testenv:coverage]
 description = report unit test coverage
-deps =
+allowlist_externals=
+    poetry
     coverage
-    pytest
-    genbadge[coverage]
+    genbadge
+commands_pre =
+    poetry install --with dev
```

Comment:
```suggestion
    poetry install --extras dev
```

Reply:
Thank you, I missed changing these changes in tox.ini file!